### PR TITLE
Upgrade editor packages

### LINKED ISSUE #48870: Completion tests fail after updating Editor to 16.8.181
See test results https://dev.azure.com/dnceng/public/_build/results?buildId=863202&view=ms.vss-test-web.build-test-results-tab&runId=27649786&resultId=166437&paneView=debug for https://github.com/dotnet/roslyn/pull/48866.

Blocking move of EditorFeatures.Cocoa to Roslyn

### PR DESCRIPTION
**Pull request by genlu**
Fix #48870

New version of editor packages are released in 16.9p4

### CODE REVIEW

File: src/Compilers/Extension/Roslyn.Compilers.Extension.csproj [hunk 1]
```diff
@@ -122,5 +122,6 @@
     <PackageReference Include="Microsoft.VisualStudio.Threading" Version="$(MicrosoftVisualStudioThreadingVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.Validation" Version="$(MicrosoftVisualStudioValidationVersion)" />
     <PackageReference Include="Newtonsoft.Json" Version="$(NewtonsoftJsonVersion)" />
+    <PackageReference Include="Microsoft.ServiceHub.Framework" Version="$(MicrosoftServiceHubFrameworkVersion)" />
```
Comment (jasonmalinowski) [inline src/Compilers/Extension/Roslyn.Compilers.Extension.csproj:125]: Keep the PackageReferences sorted?

File: src/ExpressionEvaluator/Package/ExpressionEvaluatorPackage.csproj [hunk 2]
```diff
@@ -89,5 +89,6 @@
     <PackageReference Include="Microsoft.VisualStudio.Text.Data" Version="$(MicrosoftVisualStudioTextDataVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.Threading" Version="$(MicrosoftVisualStudioThreadingVersion)" />
     <PackageReference Include="Newtonsoft.Json" Version="$(NewtonsoftJsonVersion)" />
+    <PackageReference Include="Microsoft.ServiceHub.Framework" Version="$(MicrosoftServiceHubFrameworkVersion)" />
```
Comment (jasonmalinowski) [inline src/ExpressionEvaluator/Package/ExpressionEvaluatorPackage.csproj:92]: Keep the PackageReferences sorted?

File: src/VisualStudio/Setup.Dependencies/Roslyn.VisualStudio.Setup.Dependencies.csproj [hunk 3]
```diff
@@ -58,6 +58,7 @@
     <PackageReference Include="System.Text.Encoding.CodePages" Version="$(SystemTextEncodingCodePagesVersion)" />
     <PackageReference Include="System.Threading.Tasks.Extensions" Version="$(SystemThreadingTasksExtensionsVersion)" />
     <PackageReference Include="Newtonsoft.Json" Version="$(NewtonsoftJsonVersion)" />
+    <PackageReference Include="Microsoft.ServiceHub.Framework" Version="$(MicrosoftServiceHubFrameworkVersion)" />
```
Comment (jasonmalinowski) [inline src/VisualStudio/Setup.Dependencies/Roslyn.VisualStudio.Setup.Dependencies.csproj:61]: Keep the PackageReferences sorted?

**Review by jasonmalinowski — APPROVED** [review]

**Comment by jasonmalinowski** [comment]
@genlu: Do you plan on merging this soon, as it might be a bit before we have a 16.9 VS image that we can run integration tests on.

**Comment by genlu** [comment]
@jasonmalinowski There's no hurry on this, but I do want to run our unit tests with newer version here to make sure the editor regression is fixed.

File: src/Compilers/Extension/Roslyn.Compilers.Extension.csproj [hunk 4]
```diff
@@ -109,12 +109,11 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="EnvDTE" Version="$(EnvDTEVersion)" />
+    <PackageReference Include="Microsoft.ServiceHub.Framework" Version="$(MicrosoftServiceHubFrameworkVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.ProjectSystem.Managed" Version="$(MicrosoftVisualStudioProjectSystemManagedVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.Shell.15.0" Version="$(MicrosoftVisualStudioShell150Version)" />
     <PackageReference Include="Microsoft.VisualStudio.Shell.Interop.8.0" Version="$(MicrosoftVisualStudioShellInterop80Version)" />
     <PackageReference Include="Microsoft.VisualStudio.Shell.Interop.10.0" Version="$(MicrosoftVisualStudioShellInterop100Version)" />
-    <!-- Work around https://devdiv.visualstudio.com/DevDiv/_workitems/edit/725508 by referencing Microsoft.VisualStudio.Text.Data -->
```
Comment (genlu) [inline src/Compilers/Extension/Roslyn.Compilers.Extension.csproj:116]: This is fixed.

File: src/Setup/DevDivInsertionFiles/DevDivInsertionFiles.csproj [hunk 5]
```diff
@@ -115,13 +115,6 @@
       Exclude dependencies that are not inserted from Roslyn repo and are installed by other VS components.
       
       Note: can't use globs due to https://github.com/Microsoft/msbuild/issues/3791.
-      
-      these dependency should be removed once StreamJsonRpc adds a build target for
-      net472 - https://github.com/dotnet/roslyn/issues/34558
```
Comment (genlu) [inline src/Setup/DevDivInsertionFiles/DevDivInsertionFiles.csproj:120]: https://github.com/dotnet/roslyn/issues/34558 is fixed

**Comment by genlu** [comment]
Previous regression is fixed, but it seems there's new regression with new editor packages. FYI @AmadeusW

**Comment by genlu** [comment]
@JoeRobich I updated the PR to reference 16.9p4 bits. Not sure if it's OK to merge this now w/o breaking anything.

**Comment by JoeRobich** [comment]
> @JoeRobich I updated the PR to reference 16.9p4 bits. Not sure if it's OK to merge this now w/o breaking anything.

16.9p4 will be rolled out to the scouting queue on the 17th. Assuming there are no breaks we could move to that queue and take this change then.

File: eng/Versions.props [hunk 6]
```diff
@@ -112,26 +112,26 @@
...
     <MicrosoftVisualStudioSDKEmbedInteropTypesVersion>15.0.36</MicrosoftVisualStudioSDKEmbedInteropTypesVersion>
     <MicrosoftVisualStudioEditorVersion>$(VisualStudioEditorPackagesVersion)</MicrosoftVisualStudioEditorVersion>
-    <MicrosoftVisualStudioGraphModelVersion>16.0.28226-alpha</MicrosoftVisualStudioGraphModelVersion>
-    <MicrosoftVisualStudioImageCatalogVersion>16.8.30406.65-pre</MicrosoftVisualStudioImageCatalogVersion>
-    <MicrosoftVisualStudioImagingVersion>16.8.30406.65-pre</MicrosoftVisualStudioImagingVersion>
-    <MicrosoftVisualStudioImagingInterop140DesignTimeVersion>16.8.30406.65</MicrosoftVisualStudioImagingInterop140DesignTimeVersion>
+    <MicrosoftVisualStudioGraphModelVersion>16.9.30921.310</MicrosoftVisualStudioGraphModelVersion>
+    <MicrosoftVisualStudioImageCatalogVersion>16.9.30921.310</MicrosoftVisualStudioImageCatalogVersion>
+    <MicrosoftVisualStudioImagingVersion>16.9.30921.310</MicrosoftVisualStudioImagingVersion>
+    <MicrosoftVisualStudioImagingInterop140DesignTimeVersion>16.9.30921.310</MicrosoftVisualStudioImagingInterop140DesignTimeVersion>
```
Comment (JoeRobich) [inline eng/Versions.props:134]: Would it be worthwhile setting these to the ShellPackagesVersion as well?

File: eng/Versions.props [hunk 7]
```diff
@@ -48,19 +48,19 @@
...
     <ICSharpCodeDecompilerVersion>6.1.0.5902</ICSharpCodeDecompilerVersion>
-    <MicrosoftBuildVersion>15.3.409</MicrosoftBuildVersion>
-    <MicrosoftBuildFrameworkVersion>15.3.409</MicrosoftBuildFrameworkVersion>
+    <MicrosoftBuildVersion>16.5.0</MicrosoftBuildVersion>
+    <MicrosoftBuildFrameworkVersion>16.5.0</MicrosoftBuildFrameworkVersion>
     <MicrosoftBuildLocatorVersion>1.2.6</MicrosoftBuildLocatorVersion>
-    <MicrosoftBuildRuntimeVersion>15.3.409</MicrosoftBuildRuntimeVersion>
-    <MicrosoftBuildTasksCoreVersion>15.3.409</MicrosoftBuildTasksCoreVersion>
+    <MicrosoftBuildRuntimeVersion>16.5.0</MicrosoftBuildRuntimeVersion>
+    <MicrosoftBuildTasksCoreVersion>16.5.0</MicrosoftBuildTasksCoreVersion>
```
Comment (JoeRobich) [inline eng/Versions.props:60]: Should we introduce a `MicrosoftBuildPackagesVersion`?

Comment (genlu) [reply eng/Versions.props:60]: Done.

File: eng/Versions.props (hunk 6 above)
Comment (genlu) [reply eng/Versions.props]: Sure, if they becomes out of sync in the future, we can undo it.

**Comment by genlu** [comment]
@JoeRobich I have a validation build running now. I assume this can be safely merged if CI and RPS are all green?

**Review by JoeRobich — APPROVED** [review]

**Comment by JoeRobich** [comment]
> @JoeRobich I have a validation build running now. I assume this can be safely merged if CI and RPS are all green?

Typically, would be fine. We aren't inserting master-vs-deps at this time. 

@RikkiGibson You see any need to wait on this one?

**Comment by genlu** [comment]
@JoeRobich RPS passed https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_git/VS/pullrequest/305303

**Comment by genlu** [comment]
@JoeRobich Hmm, seems CI stuck even though all integration tests legs are green.

**Comment by JoeRobich** [comment]
CI is passing. RPS is passing. Merging this PR.

**Pull request merged**